setup-env

```diff
@@ -39,6 +39,44 @@ python_versions() {
   pyenv versions --bare --skip-aliases --skip-envs
 }
 
+check_python_version() {
+  local version=$1
+
+  # Break down the regex into readable parts major.minor.patch
```

Comment:
Can you please include a comment linking to https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string to show that this is a valid regex for semantic version strings?

Reply:
I am fine with this but I should mention that though my semantic structure is based from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string I actually had to make some tweaks to it to get it working correctly. It was a while ago so I would have to figure out what tweaks I had changed specifically.

Reply:
Sounds good- that should all be documented in a comment, including why you made those tweaks.

Reply:
I'll wait for this comment to review.  I am surprised tweaks were required.

Reply:
Well after going through my notes I wasn't able to reproduce the same error that I was receiving before. Which seemed like it was just saying any version I was giving it was invalid semantically. I suspect it might have been an issue with the test that I was using to run it. That being said my change was very minor:

```bash
local suggested_regex="^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"

local tweaked_regex="^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?\$"
```
Since the suggested version actually works I am just going to use that one instead. Getting the change in now.

Reply:
@dav3r see here: https://github.com/cisagov/skeleton-generic/pull/176/commits/42ef8c2d7b54cde82d4390a0050622cddfccf92a

Reply:
I'd prefer that we include the more authoritative link to semver.org which contains that link to regex101.com.

Reply:
Looks good now, thanks!